src/main/java/org/example/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+package org.example.bookstore.service.impl;
+
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.example.bookstore.dto.order.CreateOrderRequestDto;
+import org.example.bookstore.dto.order.OrderResponseDto;
+import org.example.bookstore.dto.order.UpdateOrderRequestDto;
+import org.example.bookstore.dto.orderitem.OrderItemResponseDto;
+import org.example.bookstore.exception.EntityNotFoundException;
+import org.example.bookstore.mapper.OrderItemMapper;
+import org.example.bookstore.mapper.OrderMapper;
+import org.example.bookstore.model.Order;
+import org.example.bookstore.model.OrderItem;
+import org.example.bookstore.model.ShoppingCart;
+import org.example.bookstore.model.User;
+import org.example.bookstore.repository.order.OrderRepository;
+import org.example.bookstore.repository.orderitem.OrderItemRepository;
+import org.example.bookstore.repository.shoppingcart.ShoppingCartRepository;
+import org.example.bookstore.service.OrderService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    @Transactional
+    public OrderResponseDto createOrder(CreateOrderRequestDto requestDto) {
+        User user = getAuthenticatedUser();
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId());
+
+        Order order = new Order();
+        order.setUser(user);
+        order.setStatus(Order.Status.PROCESSING);
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(requestDto.getShippingAddress());
+
+        Set<OrderItem> orderItems = orderItemMapper.toOrderItemSet(shoppingCart.getCartItems());
+        orderItems.forEach(o -> o.setOrder(order));
```

Comment:
for each is terminal operation `orderItems.forEach(o -> o.setOrder(order));`
Check please that you don't get an exception when you try to use 
`orderItems.forEach(o -> o.setPrice(o.getPrice().multiply(new BigDecimal(o.getQuantity()))));`

Reply:
Also, rename o

Reply:
I checked, works without errors

Reply:
for each is terminal operation orderItems.forEach(o -> o.setOrder(order));
Yes, but I'm changing the value in an already created object, so I think it's okay. Correct it if it's wrong, please